tools/pastel-utility/cmd/init.go

```diff
@@ -0,0 +1,307 @@
+		return runWalletSubCommand(ctx, config)
+	})
+
+	supernodeSubCommand := cli.NewCommand("supernode")
+	supernodeSubCommand.SetUsage("Perform Supernode/Masternode specific initialization")
+	supernodeSubCommand.SetActionFunc(func(ctx context.Context, args []string) error {
+		return runSuperNodeSubCommand(ctx, config)
+	})
+
+	initCommand.AddSubcommands(
+		walletnodeSubCommand,
+		supernodeSubCommand,
+	)
+
+	initCommand.SetActionFunc(func(ctx context.Context, args []string) error {
+		ctx = log.ContextWithPrefix(ctx, "app")
+		if len(args) == 0 {
+			return errSubCommandRequired
+		}
+		return runInit(ctx, config)
+	})
+
+	return initCommand
+}
+
+func runInit(ctx context.Context, config *configs.Config) error {
+	log.WithContext(ctx).Info("Init")
+	defer log.WithContext(ctx).Info("End")
+
+	log.WithContext(ctx).Infof("Config: ")
+	log.WithContext(ctx).Infof(config.String())
+
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	sys.RegisterInterruptHandler(cancel, func() {
+		log.WithContext(ctx).Info("Interrupt signal received. Gracefully shutting down...")
+		os.Exit(0)
+	})
+
+	//run the init command logic flow
+	err := initCommandLogic(ctx, config)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// initCommandLogic runs the init command logic flow
+// takes all provided arguments in the cli and does all the background tasks
+// Print success info log on successfully ran command, return error if fail
+func initCommandLogic(ctx context.Context, config *configs.Config) error {
+	zksnarkPath := filepath.Join(config.WorkDir, "/.pastel-params/")
+
+	if _, err := os.Stat(zksnarkPath); os.IsNotExist(err) || config.Init.Force {
+		if err := os.MkdirAll(zksnarkPath, os.ModePerm); err != nil {
+			return errors.Errorf("could not create zksnarkpath %s, %v", zksnarkPath, err)
+		}
+	}
+
+	v := viper.New()
+	v.SetConfigType("json")
+	if err := v.Unmarshal(config); err != nil {
+		return err
+	}
+
+	if config.Init.Force {
+		if err := v.WriteConfigAs(defaultConfigFile); err != nil {
+			return err
+		}
+	} else {
+		if err := v.SafeWriteConfigAs(defaultConfigFile); err != nil {
+			return err
+		}
+	}
+
+	// download zksnark params
+	if err := downloadZksnarkParams(ctx, zksnarkPath, config.Init.Force); err != nil {
+		return err
+	}
+
+	return checkLocalAndRouterFirewalls(ctx, []string{"80", "21"})
+}
+
+// downloadZksnarkParams downloads zksnark params to the specified forlder
+// Print success info log on successfully ran command, return error if fail
+func downloadZksnarkParams(ctx context.Context, path string, force bool) error {
+	log.WithContext(ctx).Infoln("Downloading zksnark files")
+
+	for _, zksnarkParamsName := range configs.ZksnarkParamsNames {
+		zksnarkParamsPath := filepath.Join(path, zksnarkParamsName)
+
+		if _, err := os.Stat(zksnarkParamsPath); os.IsNotExist(err) || force {
+			out, err := os.Create(zksnarkParamsPath)
+			if err != nil {
```

Comment:
```suggestion
		if _, err := os.Stat(zksnarkParamsPath); err == nil && !force {
			log.WithContext(ctx).Infof("zksnark file %s already exists and force set to %v", zksnarkParamsPath, force)
			continue
		}
		
		out, err := os.Create(zksnarkParamsPath)
		if err != nil {
		....
```

Write simpler, no need to nest large chunks of code deeper.